src/js/page.js

```diff
@@ -89,6 +89,22 @@ export default {
         // If $e is a comment question, and it is not inside a group, there may be no $closest.
         if ( $closest.length ) {
             this._flipTo( $closest[ 0 ] );
+        } else {
+            let $sibling = $e[ 0 ].previousElementSibling;
+            let refererName = $e[ 0 ].querySelector( 'input' ).getAttribute( 'data-for' );
+            while ( $sibling ) {
+                if ( $sibling.matches( '[role="page"]' ) ) {
+                    let $input = $sibling.querySelector( '[name="' + refererName + '"]' );
+                    if ( $input ) {
+                        this._flipTo( $sibling );
+                        $sibling = null;
+                    } else {
+                        $sibling = $sibling.previousElementSibling;
+                    }
+                } else {
+                    $sibling = $sibling.previousElementSibling;
+                }
+            }
```

Comment:
This fix relies on placing comment questions as siblings of the questions they link to, which is recommended but not obligatory. 

I think the determination of refererName is too fragile, and may easily cause an exception.

Consider directly looking for the linked question, e.g with something like this:
```js
const e = $e[0];// we're trying to get rid of jQuery, so let's not add any more. Feel free to rewrite some additional existing usage in this function (e.g. let closest = e.closest('[role="page"]') etc.
...
else {
  // if $e is a comment question, and it is not inside a group, there is no closest page
   const referer = e.querySelector('input[data-for]');
   const ancestor = e.closest('.or-repeat, form.or');
   if (referer && ancestor){
     const linkedQuestion = ancestor.querySelector(`[name="${refer.dataset.for}"]`);
     if (linkedQuestion){
        this._flipTo($(linkedQuestion));
     }
  }
}
```

I also needs a clear test (for the regular comment widget, not the OC dn widget). Otherwise, this fix cannot go in Enketo Core. See test/forms/comment.xml for an example form that you can perhaps use as-is, or copy and augment.

Reply:
@MartijnR ~ code updated and tested on enketo-express-oc. Is that acceptable?
Also, I'm not sure about the test case. I test the default one on the test directory(http://localhost:8005/?xform=comment.xml) it still works fine also runs npm test. 
you said 
> See test/forms/comment.xml for an example form that you can perhaps use as-is, or copy and augment.

you mean to add a new test on the widget.comment.spec.js, right? if yes, then what kind of test should I add?
Thanks for the help

Reply:
I think the test should probably go in page.spec.js. There are different ways of doing this, but of course make sure the test fails without your fix.

One way is to call flipToPageContaining manually, and then check if the current page has been updated (or that _flipTo was called with the correct parameter).

Reply:
got it, test-case added and I update the page.js by removing the input tag on query selector because if the user sets the input comment appearance to multiline then it will be generated as textarea, not input so the selector will be failed if updated.